vision-quickstart/app/src/main/java/com/google/mlkit/vision/demo/VisionProcessorBase.java

```diff
@@ -164,15 +169,22 @@ public void processImageProxy(ImageProxy image, GraphicOverlay graphicOverlay) {
             bitmap = BitmapUtils.getBitmap(image);
         }
 
-        requestDetectInImage(
-                InputImage.fromMediaImage(image.getImage(), image.getImageInfo().getRotationDegrees()),
-                graphicOverlay,
-                /* originalCameraImage= */ bitmap,
-                /* shouldShowFps= */ true)
-                // When the image is from CameraX analysis use case, must call image.close() on received
-                // images when finished using them. Otherwise, new images may not be received or the camera
-                // may stall.
-                .addOnCompleteListener(results -> image.close());
+        if(bufferImage.size()>depth)
+            bufferImage.clear();        // O(n) operation
+
+        bufferImage.add(new Pair(image.getImage(), System.currentTimeMillis()));
```

Comment:
time difference between consecutive buffer addition takes around 20-40 ms.

Reply:
isn't this just the time between new frames from the camera?  You aren't doing anything with the images here.

Reply:
@mwarton I expected it that way and it should have been around ~16 ms for a 60fps camera. I wanted to test the processing dealay between two consecutive buffer addition. Contrary to my expectation it was around 20-40 ms. Could be due to memory copy operations.